File: wayland/src/main/java/org/westmalle/wayland/nativ/libxcb/xcb_client_message_data_t.java
//Copyright 2015 Erik De Rijcke
//
//Licensed under the Apache License,Version2.0(the"License");
//you may not use this file except in compliance with the License.
//You may obtain a copy of the License at
//
//http://www.apache.org/licenses/LICENSE-2.0
//
//Unless required by applicable law or agreed to in writing,software
//distributed under the License is distributed on an"AS IS"BASIS,
//WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,either express or implied.
//See the License for the specific language governing permissions and
//limitations under the License.
package org.westmalle.wayland.nativ.libxcb;

import com.github.zubnix.jaccall.CType;
import com.github.zubnix.jaccall.Field;
import com.github.zubnix.jaccall.Struct;

@Struct(union = true,
        value = {
                @Field(name = "data8",
                       type = CType.CHAR,
                       cardinality = 20),
                @Field(name = "data16",
                       type = CType.SHORT,
                       cardinality = 10),
                @Field(name = "data32",
                       type = CType.INT,
                       cardinality = 5),
        })
public final class xcb_client_message_data_t extends xcb_client_message_data_t_Jaccall_StructType {}

File: compositor/src/main/kotlin/org/westford/nativ/libxcb/xcb_client_message_data_t.kt
/*
 * Westford Wayland Compositor.
 * Copyright (C) 2016  Erik De Rijcke
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as
 * published by the Free Software Foundation, either version 3 of the
 * License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */
package org.westford.nativ.libxcb

import org.freedesktop.jaccall.CType
import org.freedesktop.jaccall.Field
import org.freedesktop.jaccall.Struct

@Struct(union = true,
        value = *arrayOf(Field(name = "data8",
                               type = CType.CHAR,
                               cardinality = 20),
                         Field(name = "data16",
                               type = CType.SHORT,
                               cardinality = 10),
                         Field(name = "data32",
                               type = CType.INT,
                               cardinality = 5))) class xcb_client_message_data_t : Struct_xcb_client_message_data_t()
